docs/dev_cycle.part-01.rst

```diff
@@ -0,0 +1,242 @@
+If the ETH rate is rising, then we will probably see smaller donations
+in terms of Ether, but similar donations in terms of USD.
+
+In other words, two donations of say $30 will have different amounts in ETH if the exchange rate changed between
+the donations. As a simple solution, we will ask donators to provide the effective ETH/USD exchange rate when they send their (hopefully generous) donations.
+
+Here is the new contract code:
+
+.. code-block:: solidity
+
+    pragma solidity ^0.4.0;
+
+    /// TUTORIAL CONTRACT DO NOT USE IN PRODUCTIO
+    /// @title Donations collecting contract
+
+    contract Donator {
+
+      uint public donations_total;
+      uint public donations_usd;
+      uint public donations_count;
+      uint public default_usd_rate;
+
+      function Donator() {
+        default_usd_rate = 350;
+      }
+      modifier money_sent() { if (!msg.value > 0) throw; _; }
+      function donate(uint usd_rate) public payable money_sent {
+          donations_total += msg.value;
+          donations_count += 1;
+          default_usd_rate = usd_rate;
+          uin in_usd = msg.value * usd_rate;
+          donations_usd += in_usd;
+      }
+      // fallback function
+      function () payable {
+        donate(default_usd_rate);
+      }
+     }
+
+
+
+Save the code to ``contracts/Donator.sol``.
+
+
+Quick Solidity Overview 
+-----------------------
+
+**Pragma**:
+Every Solidity source should provide the compiler compatability: `pragma solidity ^0.4.0;`
+
+**Contract definition**:
+The ``contract`` keyword starts a new contract definition, named ``Donator``. 
+
+.. note::
+
+    Contracts names should follow class naming rules (like MyWallet, GoodLuck or WhyNot).
+
+**State variables**:
+The contract has 4 state variables: ``donations_total``, ``donations_usd``, ``donations_count`` and ``default_usd_rate``.
+A state variable is defined in the *contract scope*.
+State variables are saved in the contract's persisten *storage*, 
+kept after the transaction run ends, and synced to every node on the blockchain.
+
+**Visibility:**
+The ``public`` decleration ensures that all state variables and the ``donate`` function will be available for the callers
+of the contrat, in the contract's interface.
+
+.. note::   
+    For the public state variables, the compiler actually creates an accessor function
+    which if you had to type manually could look like: ``function total() public returns (uint) {return donations_total;}``
+
+**Data types**:
+Since we are dealing with numbers, the only data type we use here is ``uint``, unsigned integer. The ``int`` and ``uint``
+are declated in steps of 8 bits, ``unint8``, ``uint16`` etc. When the bits indicator is omitted, like ``int`` or ``uint``, the compiler will
+assumes ``uint256``.
+
+.. note:: 
+
+    If you know in advance the the maximum size of a variable, better to limit the type and save the gas of extra 
+    memory or storage.
```

Comment:
This is not entirely accurate.  Solidity only saves memory and storage in certain situations, and often, using smaller types means higher gas costs because solidity has to do the conversions.